crypto/schemes/enc/v1/README.md

```diff
@@ -0,0 +1,137 @@
+# Dapr encryption scheme v1: `dapr.io/enc/v1`
+
+This document contains the reference for the **Dapr encryption scheme v1**, identified by the header `dapr.io/enc/v1`.
+
+The Dapr encryption scheme is optimized for processing data as a stream. Data is chunked into multiple parts which are encrypted independently. This allows us to return data to callers as a stream, even when decrypting messages, being confident that we are not flushing unverified data to the client.
+
+> **Sources:** The encryption scheme that Dapr uses is heavily inspired by the [Tink wire format](https://developers.google.com/tink/wire-format) (from the Tink library maintained by Google), as well as by Filippo Valsorda's [age](https://age-encryption.org/v1), and Minio's [DARE](https://github.com/minio/sio).
+
+## Key
+
+Each message is encrypted with a 256-bit symmetric **File Key (FK)** that is randomly generated by Dapr for each new message. The key must be generated as 32 byte of output from a CSPRNG (such as Go's `crypto/rand.Reader`) and must not be reused for other files.
+
+The FK is wrapped using a key stored in a key vault (**Key Encryption Key (KEK)**) by Dapr. The result of the wrapping operation is the **Wrapped File Key (WFK)**. The algorithm used depends on the type of the KEK as well as the algorithms supported by the component:
+
+- For symmetric keys:
+    - AES-KW ([RFC 3394](https://www.rfc-editor.org/rfc/rfc3394.html)): `AES-KW`
+- For RSA keys:
+    - RSA OAEP with SHA-256: `RSA-OAEP-256`
+
+> Other key wrapping algorithms can be implemented in the future.
+
+## Ciphertext format
+
+The ciphertext is formatted as:
+
+```text
+header || binary_payload
+```
+
+## Header
+
+The **header** is human-readable and contains 3 items, each terminated by a line feed (`0x0A`) character:
+
+1. Name and version of the encryption scheme used. In this version of the spec, this is always `dapr.io/enc/v1`.
+2. The manifest, which is a JSON object.
+3. The MAC for the header, base64-encoded.
+
+> Base64 encoding follows [RFC 4648 §4](https://datatracker.ietf.org/doc/html/rfc4648#section-4) ("standard" format, with padding included)
+
+For example:
+
+```text
+dapr.io/enc/v1
+{"k":"mykey","kw":1,"wfk":"hGYjwDpWEXEymSTFZ95zgX8krElb3Gqyls67R8zJA3k=","cph":1,"np":"Y3J5cHRvIQ=="}
+pBDKLrhAWL7IAvDKBV/v7lmbTG6AEZbf3srUN0Pnn30=
+```
+
+### Manifest
+
+The second line in the header is the **manifest**, which is a compact JSON object.
+
+Its corresponding Go struct is:
+
+```go
+type Manifest struct {
+	// Name of the key that can be used to decrypt the message.
+	// This is optional, and if specified can be in the format `key` or `key/version`.
+	KeyName string `json:"k,omitempty"`
+	// ID of the wrapping algorithm used.
```

Comment:
We do not define 0x00 here or for WFK. Is this intentional to leave anything else as undefined?

Reply:
Sorry I missed this.

Not defining value `0` is on purpose, so if that were to be encountered it would be considered an error. Otherwise, it'd be harder to distinguish value missing vs intentionally setting this to the zero value.